examples/dcmip_3_1_meanflow_quads.py

```diff
@@ -164,19 +164,31 @@
 advected_fields.append(("theta", SSPRK3(state, theta0, thetaeqn, subcycles=2)))
 
 # Set up linear solver
-solver_parameters = {
-    'ksp_type': 'fgmres',
-    'ksp_rtol': 1.0e-6,
-    'ksp_max_it': 100,
-    'pc_type': 'gamg',
-    'mg_levels': {'ksp_type': 'gmres',
-                  'ksp_max_it': 8,
-                  'pc_type': 'bjacobi',
-                  'sub_pc_type': 'ilu'}
-}
+# if not gamg then use basic ilu
+gamg = False
+if gamg:
+    solver_parameters = {
+        'ksp_type': 'fgmres',
+        'ksp_monitor': True,
+        'ksp_rtol': 1.0e-6,
+        'ksp_max_it': 100,
+        'pc_type': 'gamg',
+        'pc_gamg_sym_graph': True,
+        'mg_levels': {'ksp_type': 'gmres',
+                      'ksp_max_its': 8,
+                      'pc_type': 'bjacobi',
+                      'sub_pc_type': 'ilu'}
+    }
```

Comment:
Out of curiosity: is the issue that the solver fails to converge for this example, or does it just take too many iterations?